packages/api/src/external/commonwell/patient-shared.ts

```diff
@@ -251,9 +251,13 @@ export async function searchPersons({
 }
 
 export function getPersonalIdentifiersFromPatient(patient: Patient): SimplifiedPersonalId[] {
-  return (patient.data.personalIdentifiers ?? []).flatMap(id =>
-    id.value !== undefined && id.state !== undefined
-      ? { key: id.value, system: driversLicenseURIs[id.state] }
-      : []
-  );
+  return (patient.data.personalIdentifiers ?? [])
+    .filter(id => id.type === "driversLicense") // Skip non-driversLicense for CW
+    .flatMap(id =>
+      id.value !== undefined && id.type === "driversLicense" && id.state !== undefined
+        ? { key: id.value, system: driversLicenseURIs[id.state] }
+        : id.value !== undefined && id.type !== "driversLicense"
+        ? { key: id.value, system: identifierSytemByType[id.type] }
+        : []
+    );
```

Comment:
Since we're also filtering on the `flatMap`, what do you think we merge lines 255 and 257 into a single function, like `isValidDriverLicense()`?

Could also rewrite this logic to something more easier to read, w/ a default value at the end, like:

```ts
.flatMap(id => {
  if (isValidDriversLicense(id)) return getDriverLicensePersonalIdentifier(id);
  if (id.value !== undefined) return getGeneralPersonalIdentifier(id);
  return [];
});
```

Reply:
Rewrote this in `patient-conversion.ts`. Didn't go full `isValidDriversLicense` as if you look at the typing for `PersonalIdentifiers` on `Patient`, `value` is always set, so this function is essentially too complicated anyway and it boils down to a simply checking `type` (which doesn't feel necessary to make a helper function yet).